src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java

```diff
@@ -11,61 +11,111 @@
-                    () -> zookeeperBrokers + zookeeperKafkaNamespace);
-            final ExhibitorRestClient exhibitorRestClient = new DefaultExhibitorRestClient();
-            ensembleProvider = new ExhibitorEnsembleProvider(exhibitors,
-                    exhibitorRestClient, "/exhibitor/v1/cluster/list", EXHIBITOR_POLLING_MS, retryPolicy);
-            ((ExhibitorEnsembleProvider) ensembleProvider).pollForInitialEnsemble();
+    public CuratorFramework get() {
+        return curator;
+    }
+
+    /**
+     * Parses string in form: (exhibitor|zookeeper):(hostname)[(,(hostname))*]:(port)/(root_path)
+     * to parse connection string. For example:
+     * <p>
+     * exhibitor://ex1.zalando.de,ex2.zalando.de:8181/nakadi
+     * - specifies that list of servers should be taken by exhibitor protocol on port 8181, instances
+     * 127.0.0.1, 127.0.0.2, and root path is /nakadi
+     * <p>
+     * zookeeper://ex1.zalando.de,ex2.zalando.de,ex3.zalando.de:2181/nakadi
+     * - means that there are 3 zookeepers ex1, ex2, ex3 with client port 2181, root path for curator library is
+     * /nakadi
+     *
+     * @param zooConnectionString Connection string to parse
+     * @return Zookeeper holder to use
+     */
+    public static ZooKeeperHolder build(final String zooConnectionString) {
+        final String[] protocolAndProtocolData = zooConnectionString.split("://");
+        if (protocolAndProtocolData.length != 2) {
+            throw new RuntimeException("Connection string is wrong for " + zooConnectionString);
+        }
+        final int pathLocation = protocolAndProtocolData[1].indexOf('/');
+        final String path;
+        final String serversAndPort;
+        if (pathLocation < 0) {
+            path = "/";
+            serversAndPort = protocolAndProtocolData[1];
         } else {
-            ensembleProvider = new FixedEnsembleProvider(zookeeperBrokers + zookeeperKafkaNamespace);
+            path = protocolAndProtocolData[1].substring(pathLocation);
+            serversAndPort = protocolAndProtocolData[1].substring(0, pathLocation);
         }
-        zooKeeper = CuratorFrameworkFactory.builder()
-                .ensembleProvider(ensembleProvider)
+
+        final RetryPolicy retryPolicy = new ExponentialBackoffRetry(EXHIBITOR_RETRY_TIME, EXHIBITOR_RETRY_MAX);
+
+        final CuratorFramework framework = CuratorFrameworkFactory.builder()
+                .ensembleProvider(createEnsembleProvider(protocolAndProtocolData[0], serversAndPort, path))
                 .retryPolicy(retryPolicy)
                 .build();
-        zooKeeper.start();
+
+        framework.start();
+        return new ZooKeeperHolder(framework);
     }
 
-    public CuratorFramework get() {
-        return zooKeeper;
+    private static EnsembleProvider createEnsembleProvider(
+            final String type, final String serversAndPort, final String path) {
+        final EnsembleProvider result;
+        if (TYPE_EXHIBITOR.equalsIgnoreCase(type)) {
```

Comment:
why do we still need all this exhibitor stuff here? For migration period?

Reply:
Yep

Reply:
And in case if we will still use it